lib/rtcpeerconnection/chrome.js

```diff
@@ -117,12 +124,41 @@ function ChromeRTCPeerConnection(configuration) {
 
 inherits(ChromeRTCPeerConnection, EventTarget);
 
-ChromeRTCPeerConnection.prototype.addStream = function addStream(stream) {
-  if (this._localStreams.has(stream)) {
-    return;
+ChromeRTCPeerConnection.prototype.addTrack = function addTrack() {
+  var args = [].slice.call(arguments);
+  if ('addTrack' in PeerConnection.prototype) {
+    return this._peerConnection.addTrack.apply(this._peerConnection, args);
+  }
+  var mediaStreamTrack = args[0];
+  var syncMediaStreams = args.slice(1);
+
+  if (this._peerConnection.signalingState === 'closed') {
+    throw new Error('Cannot add MediaStreamTrack [' + mediaStreamTrack.id + ', '
+      + mediaStreamTrack.kind + ']: RTCPeerConnection is closed');
   }
-  this._localStreams.add(stream);
-  this._peerConnection.addStream(stream);
+  if (this._tracksToSyncStreams.has(mediaStreamTrack.id)) {
+    throw new Error('Cannot add MediaStreamTrack [' + mediaStreamTrack.id + ', '
+      + mediaStreamTrack.kind + ']: RTCPeerConnection already has it');
+  }
+
+  var syncMediaStreamIds = new Set(syncMediaStreams.map(function(mediaStream) {
+    return mediaStream.id;
+  }));
+  this._tracksToSyncStreams.set(mediaStreamTrack.id, syncMediaStreamIds);
+
+  syncMediaStreamIds.forEach(function(mediaStreamId) {
+    var containerMediaStream = this._syncStreamsToContainerStreams.get(mediaStreamId);
+    if (containerMediaStream) {
+      this._peerConnection.removeStream(containerMediaStream);
+    } else {
+      containerMediaStream = new MediaStream();
+    }
+    containerMediaStream.addTrack(mediaStreamTrack);
+    this._peerConnection.addStream(containerMediaStream);
+    this._syncStreamsToContainerStreams.set(mediaStreamId, containerMediaStream);
+  }, this);
+
+  return { track: mediaStreamTrack };
```

Comment:
Might make sense to shim RTCRtpSender out a bit here, rather than return an ad-hoc object?

Reply:
@markandrus OK.